app/scripts/modules/core/src/managed/resources/useGetResourceLink.hook.ts

```diff
@@ -0,0 +1,18 @@
+import { useSref } from '@uirouter/react';
+
+import { IResourceLinkProps, resourceManager } from './resourceRegistry';
+
+export const useGenerateLink = (props: IResourceLinkProps): { href: string; isExternal?: boolean } | undefined => {
+  const externalLink = resourceManager.getExperimentalDisplayLink(props);
+  const routingInfo = resourceManager.getNativeResourceRoutingInfo(props) ?? { state: '', params: {} };
+  const routeProps = useSref(routingInfo.state, routingInfo.params);
+
+  if (externalLink) {
+    return { href: externalLink, isExternal: true };
+  }
+
+  if (routeProps.href) {
+    return { href: routeProps.href };
```

Comment:
recommend you use the onClick handler returned from useSref

Reply:
Sure. Why?